Parse/PFObject.m

```diff
@@ -1110,7 +1110,7 @@ - (void)mergeFromRESTDictionary:(NSDictionary *)object withDecoder:(PFDecoder *)
                 if (localOperationSet.updatedAt != nil &&
                     [localOperationSet.updatedAt compare:remoteOperationSet.updatedAt] != NSOrderedAscending) {
                     [localOperationSet mergeOperationSet:remoteOperationSet];
-                } else {
+                } else if (remoteOperationSet) {
```

Comment:
This one is actually very tricky, since `remoteOperationSet` should always be here, according to the logic above.
Maybe replace it with `PFConsistencyAssert` with a useful error message?
I just checked locally - it actually supresses the analyzer warning if you have an assert here.